### PR TITLE
Added a note about the usage of NotBlank

### LINKED ISSUE #10357: Email assert passes empty string
/**
     * @var string
     *
     * @ORM\Column(name="email", type="string", length=100, unique=true)
     * @Assert\Email(
	 *     message="Email is not valid",
	 *     checkMX=true,
	 *     mode="html5"
	 * )
     */
    private $email;

passing empty string as an email passed validation. var_dump of the $validator->validate( $object ); is below.

object(Symfony\Component\Validator\ConstraintViolationList)#3636 (1) {
 ["violations":"Symfony\Component\Validator\ConstraintViolationList":private]=>
 array(0) {
 }
}

### PR DESCRIPTION
**Pull request by javiereguiluz**
Fixes #4244 and fixes #10357.

### CODE REVIEW

File: reference/constraints/Email.rst [hunk 1]
```diff
@@ -88,6 +88,14 @@ Basic Usage
             }
         }
 
+.. note::
+
+    As it happens with the rest of constraints, ``null`` values and empty strings
```
Comment (ping-localhost) [inline reference/constraints/Email.rst]: How about `As with the rest of the constraints`?

**Review by ping-localhost — APPROVED** [review]
I have a tiny comment about wording, but other than that, LGTM! :+1:

**Review by ping-localhost — APPROVED** [review]

File: reference/constraints/_empty-values-are-valid.rst.inc [hunk 2]
```diff
@@ -0,0 +1,7 @@
+.. note::
+
+    As with most of the other constraints, ``null`` values and empty strings
+    are considered valid values. Otherwise, in addition to validating this value,
+    you would also be requiring it, making it impossible to be optional. That's
+    why it's common to combine this constraint with
+    :doc:`NotBlank </reference/constraints/NotBlank>`.
```
Comment (xabbuh) [inline reference/constraints/_empty-values-are-valid.rst.inc]: I thought if we should maybe reword this a bit:

> As with most of the other constraints, `null` values and empty strings are considered valid values. This is to allow them to be optional values. If the values is mandatory, a common solution is to combine this constraint with `NotBlank`.

Comment (javiereguiluz) [reply reference/constraints/_empty-values-are-valid.rst.inc]: I think that your wording is better. Could you please create a pull request for it? Thanks a lot!

Comment (xabbuh) [reply reference/constraints/_empty-values-are-valid.rst.inc]: here we go: #10418
